### DISCUSSION TITLE
Longhorn Go Client

### DISCUSSION THREAD

**Discussion by isc-akopplin** [longhorn/longhorn#3974] [Q&A]
I am trying to use the github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned/typed/longhorn/v1beta1 client to programmatically manipulate longhorn objects. However, the go.mod file contains a reference to k8s.io/apiextensions-apiserver@v0.0.0 which is unknown. How can I leverage this longhorn client? The right reference seems to be present on the master branch. Why are the release branches in this state? This would be incredibly useful if it was documented, is there a reason it is not?

**Comment by PhanLe1010** [comment]
I think the reason is your module import Longhorn manager. Then Longhorn manager module imports Kubernetes. Then Kuberntes module imports the `k8s.io/apiextensions-apiserver@v0.0.0`. The version v0.0.0 is a local version. Kubernetes also imports other components with this invalid version as well.

To fix this, you need to specify replace statements similar to what Longhorn does https://github.com/longhorn/longhorn-manager/blob/fb4e45e7e6db63cf4b15f0f755a1b37616d9e678/go.mod#L6-L30 . Please select the correct Longhorn branch/tag to see the correct replace statements
